### PR TITLE
DOC-8161 - Named Scopes & Collections example

### PR DESCRIPTION
**Pull request by maria-robobug**
This PR adds an example for named scopes and collections for KV operations.

I've also updated the following for completeness:
- The `couchbase` dependency to `^3.1.1`
- Formatting for the `kv-operations.js` file
- The `hapi.js` version we are using for `kv-operations.js` seems [unsupported](https://www.npmjs.com/package/hapi) now, so I've updated it to the [latest](https://www.npmjs.com/package/@hapi/hapi) one.

I've tried to ensure that most of the example handlers run as expected, but there are still some that seem to fail(mainly when using callback functions), see comment below.

### CODE REVIEW

**Comment by maria-robobug** [comment]
There seem to be some examples where we use callbacks, like below:

```
 const key = request.query.k ? request.query.k : docKey;
    try {
        // #tag::callback[]
        var result = await collection.get(key,
            (err, res) => {
                if (err) console.log(err);
                if (res) console.log(res);
            }
        );
        // #end::callback[]
    } catch (e) {
        console.log(e);
        return h.response(e.toString());
    }
    cas = result.cas;
    return h.response(result);
```
When I run this I seem to get an error:  `TypeError: Cannot read property 'cas' of undefined`
So it seems that for some reason the `result` variable is `undefined` and I am not sure why that is the case, do we need a different approach when passing a callback? 

Not written JS in awhile so any suggestions would be much appreciated!

File: modules/devguide/examples/nodejs/kv-operations.js [hunk 1]
```diff
@@ -329,8 +329,6 @@ async function decrementwithoptionsHandler(request, h) {
         // #end::decrementwithoptions[]
         return h.response(result);
     } catch (e) {
-        if (e.toString() === "Error: bad initial passed")
-            return (h.response("JSCBC-670 must be fixed for 'inital' to be accepted"));
```
Comment (maria-robobug) [inline modules/devguide/examples/nodejs/kv-operations.js:339]: JSCBC-670 seems to be fixed so I've removed the check.

**Comment by brett19** [comment]
The `result` your using is out-of-scope.  It's created in the try/catch block, but then used outside of it.  Should be able to just move the usage into the try block.

**Comment by maria-robobug** [comment]
As per @brett19's suggestions I've ensured that the examples use the async/await pattern where possible, keeping only one example that uses the callback pattern (as we still want to show that this is doable).

All the examples run successfully now.

**Review by RichardSmedley — APPROVED** [review]
:+1:

**Pull request merged**